hack/golangci.yaml.in

```diff
@@ -145,8 +145,8 @@ linters:
   settings: # please keep this alphabetized
     custom:
       logcheck:
-        # Installed there by hack/verify-golangci-lint.sh.
-        path: _output/local/bin/logcheck.so
+        # Built into golangci-lint by hack/verify-golangci-lint.sh.
+        type: "module"
```

Comment:
The `run.relative-path-mode: wd` was missing in the PR.

It would not have worked well in practice because it also effects the paths that get matched against suppressions (https://github.com/kubernetes/kubernetes/pull/131477#discussion_r2063499223), so let's not do this.

/close